src/components/title/show/package-filter-modal/package-filter-modal.js

```diff
@@ -50,13 +63,32 @@ const PackageFilterModal = ({
   const handleReset = () => {
     onSubmit([]);
     setSelectedOptions([]);
+    setSearchFilters({});
     toggleModal();
   };
 
   const handleSubmit = () => {
-    const newSelectedPackages = allPackages.filter(({ id: packageId }) => selectedOptions.some((({ value: optionId }) => packageId === optionId)));
-
-    onSubmit(newSelectedPackages);
+    const newSelectedPackages = allPackages
+      .filter(
+        ({ id: packageId }) => (
+          selectedOptions.length
+            ? selectedOptions.some((({ value: optionId }) => packageId === optionId))
+            : true
+        )
+      )
+      .filter(
+        ({ isSelected, providerId }) => (
+          searchFilters.selected
```

Comment:
Please split this up so there's no complex ternary checks. It's a bit difficult to understand now

Reply:
Fixed